### ISSUE TITLE
Isentropic process - lookup with P,S inputs

### ISSUE THREAD

**Issue by d-e-t** [ibell/coolprop#203]
I'm trying to model an isentropic process with air, but CP.Props() fails when looking up enthalpy given inputs P and S.  According to the docs, this is a valid lookup combination, and it does give results for some gasses, such as CO2.

```
import CoolProp.CoolProp as CP

# Pump inlet
fluid = 'Air'
h_i   = CP.Props('H','T',298.15,'P',0.1,fluid)  # Pump inlet: 100 Pa, 25'C
s_i   = CP.Props('S','T',298.15,'P',0.1,fluid)  # kJ/kg-K
m_dot = 0.000010549                             # kg/s

# Pump outlet: assume isentropic compression
h_o   = CP.Props('H','P',101.3,'S',s_i,fluid)   # Compress to 1 atm
Work  = m_dot*(h_o-h_i)                         # Isentropic pump work
```

CoolProp fails with the follow error message:

```
Traceback (most recent call last):
  File "coolprop_isentropic_pump.py", line 14, in <module>
    h_o   = CP.Props('H','P',101.3,'S',s_i,'Air')   # Compress to 1 atm
  File "CoolProp.pyx", line 257, in CoolProp.CoolProp.Props (build/temp.linux-x86_64-2.7/pyrex/CoolProp/CoolProp.cpp:21799)
  File "CoolProp.pyx", line 374, in CoolProp.CoolProp.Props (build/temp.linux-x86_64-2.7/pyrex/CoolProp/CoolProp.cpp:20135)
ValueError: Props failed ungracefully with inputs:"H",'P',1.0130000000000000e+02,'S',8.8460397211309658e+00,"Air"; please file a ticket at https://github.com/ibell/coolprop/issues
```

**Comment by ibell** [comment]
That's a pretty huge pressure ratio, perhaps the outlet temperature is too
high so it gets bad guess values in the solver.  Other gases have lower
adiabatic indices, so a lower temperature rise.

I'll take a look.  You should be able to get a good solution for these
inputs.

**Comment by ibell** [comment]
In the interim you could solve using scipy.newton (
http://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.newton.html#scipy.optimize.newton
)

**Comment by d-e-t** [comment]
Yes, it is a vacuum pump calc, so a big pressure gradient.  The scipy.newton workaround does the trick -- thank you.  Here's the working version:

```
import CoolProp.CoolProp as CP
from scipy.optimize import newton

# Pump inlet
fluid = 'Air'
h_i   = CP.Props('H','T',298.15,'P',0.1,fluid)  # Pump inlet: 100 Pa, 25'C
s_i   = CP.Props('S','T',298.15,'P',0.1,fluid)  # kJ/kg-K
m_dot = 0.000010549                             # kg/s

# Vacuum pump outlet: assume isentropic compression
P_o   = 101.3                                   # kPa
s_o   = s_i
T_o   = newton(lambda T_o_i: CP.Props('S', 'P', P_o, 'T', T_o_i, fluid)-s_o, x0=500.0)
h_o   = CP.Props('H','P', P_o, 'T', T_o, fluid)
Work  = m_dot*(h_o-h_i)                         # Isentropic pump work
```

Having it work directly with Props() would be real nice for array calcs, but this gets me going for now.

**Comment by ibell** [comment]
Fixed in 12bef51720e8f5e005134839b28e1bbdeb3e9a23 - added a relaxation factor to PS solver

**Comment by d-e-t** [comment]
Thank you.  The example code now works as follows:

```
import CoolProp.CoolProp as CP
from scipy.optimize import newton

# Pump inlet
fluid = 'Air'
h_i   = CP.Props('H','T',298.15,'P',0.1,fluid)  # Pump inlet: 100 Pa, 25'C
s_i   = CP.Props('S','T',298.15,'P',0.1,fluid)  # kJ/kg-K
m_dot = 0.000010549                             # kg/s

# Pump outlet: assume isentropic compression
P_o   = 101.3                                   # kPa
s_o   = s_i
h_o   = CP.Props('H','P',101.3,'S',s_i,fluid)   # Compress to 1 atm
Work  = m_dot*(h_o-h_i)                         # Isentropic pump work
```
